### ISSUE TITLE
why random seed is fixed?

### ISSUE THREAD

**Issue by fyears** [dmlc/xgboost#301]
There is a parameter, `seed`. I see that it has default value 0. So everytime we run the model we get the exactly same result if we do not set it?!

I am using python version.

**Comment by tqchen** [comment]
This is for the reproducible experiments,  You can manually set different seeds if you want different results

**Comment by cbjorgol** [comment]
I noticed that using a seed of -1 does not provide a random seed.  This would be useful in tuning hyper-parameters so that each iteration of grid search uses a unique seed.  This ends up being a big deal if you are using a search mechanism like bayesian optimization which would treat seed as a tuning parameter and sample more heavily on seeds that provide better lift.

I think using a randomized seed if seed is set to -1 would be useful.
